File: FootScout-Vue.WebAPI/Entities/ClubAdvertisement.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace FootScout_Vue.WebAPI.Entities
{
    public class ClubAdvertisement
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int PlayerPositionId { get; set; }

        [ForeignKey("PlayerPositionId")]
        public virtual PlayerPosition PlayerPosition { get; set; }

        [Required]
        [MaxLength(30)]
        public string ClubName { get; set; }

        [Required]
        [MaxLength(30)]
        public string League { get; set; }

        [Required]
        [MaxLength(30)]
        public string Region { get; set; }

        [Required]
        public int SalaryRangeId { get; set; }

        [ForeignKey("SalaryRangeId")]
        public virtual SalaryRange SalaryRange { get; set; }

        [Required]
        [DataType(DataType.DateTime)]
        public DateTime CreationDate { get; set; }

        [Required]
        [DataType(DataType.DateTime)]
        public DateTime EndDate { get; set; }

        [Required]
        public string ClubMemberId { get; set; }

        [ForeignKey("ClubMemberId")]
        public virtual User ClubMember { get; set; }
    }
}

File: FootScout-Vue.WebClient/src/models/interfaces/ClubAdvertisement.ts
import type { PlayerPosition } from './PlayerPosition';
import type { SalaryRange } from "./SalaryRange";
import type { UserDTO } from '../dtos/UserDTO';

export interface ClubAdvertisement {
    id: number;
    playerPositionId: number;
    playerPosition: PlayerPosition;
    clubName: string;
    league: string;
    region: string;
    salaryRangeId: number;
    salaryRange: SalaryRange;
    creationDate: string;
    endDate: string;
    clubMemberId: string;
    clubMember: UserDTO;
}